integration_tests/tests/core.rs

```diff
@@ -32,31 +25,32 @@ fn get_caller_identity_presigned() {
         .get(&url)
         .header("x-test-header", "foobar")
         .send()
+        .await
         .expect("to succeed");
     assert!(
         response.status().is_success(),
         "presigned url should succeed when used"
     );
 }
 
-#[test]
-fn with_signature() {
+#[tokio::test]
+async fn with_signature() {
     let client = Client::shared();
     let mut request = SignedRequest::new("GET", "sts", &Region::UsEast1, "/");
     let mut params = Params::new();
     params.put("Action", "GetCallerIdentity");
     params.put("Version", "2011-06-15");
     request.set_params(params);
     let response = client
-        .sign_and_dispatch::<HttpResponse, ()>(request, |r| Box::new(ok(r)))
-        .sync();
+        .sign_and_dispatch(request)
+        .await;
     assert!(response.is_ok(), response.err());
     let response: HttpResponse = response.unwrap();
     assert!(response.status == 200, format!("Signed request should succeed with status code 200. Got status code: {:?}, headers {:?}", response.status, response.headers));
 }
 
-#[test]
-fn without_signature() {
+#[tokio::test]
+async fn without_signature() {
```

Comment:
This integration test is failing.

<details><summary>Backtrace output</summary>

```
running 3 tests
test without_signature ... FAILED
test get_caller_identity_presigned ... ok
test with_signature ... ok

failures:

---- without_signature stdout ----
thread 'without_signature' panicked at 'called `Option::unwrap()` on a `None` value', src/libcore/option.rs:378:21
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:77
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:61
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1028
   5: std::io::Write::write_fmt
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libstd/io/mod.rs:1412
   6: std::io::impls::<impl std::io::Write for alloc::boxed::Box<W>>::write_fmt
             at src/libstd/io/impls.rs:141
   7: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:65
   8: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:50
   9: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:188
  10: std::panicking::default_hook
             at src/libstd/panicking.rs:202
  11: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:464
  12: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:373
  13: rust_begin_unwind
             at src/libstd/panicking.rs:302
  14: core::panicking::panic_fmt
             at src/libcore/panicking.rs:139
  15: core::panicking::panic
             at src/libcore/panicking.rs:70
  16: core::option::Option<T>::unwrap
  17: rusoto_core::client::sign_and_dispatch::{{closure}}
  18: <std::future::GenFuture<T> as core::future::future::Future>::poll::{{closure}}
  19: std::future::set_task_context
  20: <std::future::GenFuture<T> as core::future::future::Future>::poll
  21: std::future::poll_with_tls_context::{{closure}}
  22: std::future::get_task_context
  23: std::future::poll_with_tls_context
  24: <rusoto_core::client::ClientInner<P,D> as rusoto_core::client::SignAndDispatch>::sign_and_dispatch::__sign_and_dispatch::{{closure}}
  25: <std::future::GenFuture<T> as core::future::future::Future>::poll::{{closure}}
  26: std::future::set_task_context
  27: <std::future::GenFuture<T> as core::future::future::Future>::poll
  28: <core::pin::Pin<P> as core::future::future::Future>::poll
  29: std::future::poll_with_tls_context::{{closure}}
  30: std::future::get_task_context
  31: std::future::poll_with_tls_context
  32: rusoto_core::client::Client::sign_and_dispatch::{{closure}}
  33: <std::future::GenFuture<T> as core::future::future::Future>::poll::{{closure}}
  34: std::future::set_task_context
  35: <std::future::GenFuture<T> as core::future::future::Future>::poll
  36: std::future::poll_with_tls_context::{{closure}}
  37: std::future::get_task_context
  38: std::future::poll_with_tls_context
  39: core::without_signature::{{closure}}
  40: <std::future::GenFuture<T> as core::future::future::Future>::poll::{{closure}}
  41: std::future::set_task_context
  42: <std::future::GenFuture<T> as core::future::future::Future>::poll
  43: tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on
  44: tokio::runtime::Runtime::block_on::{{closure}}
  45: tokio::runtime::context::enter
  46: tokio::runtime::handle::Handle::enter
  47: tokio::runtime::Runtime::block_on
  48: core::without_signature
  49: core::without_signature::{{closure}}
  50: core::ops::function::FnOnce::call_once
  51: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/liballoc/boxed.rs:942
  52: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:78
  53: std::panicking::try
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libstd/panicking.rs:265
  54: std::panic::catch_unwind
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libstd/panic.rs:396
  55: test::run_test_in_process
             at src/libtest/lib.rs:570
  56: test::run_test::run_test_inner::{{closure}}
             at src/libtest/lib.rs:473
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


failures:
    without_signature

test result: FAILED. 2 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
```
</details>